librad/src/git/local/url.rs

```diff
@@ -0,0 +1,107 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    fmt::{self, Display},
+    str::FromStr,
+};
+
+use thiserror::Error;
+
+use crate::{
+    hash::{self, Hash},
+    uri::{self, RadUrn},
+};
+
+#[derive(Clone)]
+pub struct LocalUrl {
+    repo: Hash,
```

Comment:
Is `repo` the namespace? If yes, maybe we could name it `namespace`.

Reply:
`namespace` is just how we happen to store it, `repo` is what it is logically